### DISCUSSION TITLE
Adding Raspberry PI as a second MCU

### DISCUSSION THREAD

**Discussion by linhungsam** [OpenNeptune3D/OpenNept4une#193] [Ideas]
Hi all,

I am curios is it possible to add RPI as second MCU so adding accessories like a new LED light, a new fan, external heater (heated chamber) any thing that can be power by GPIO?

Ideally, moving all the brain power to RPI and let RPI control the micro-controller.

Sam

**Comment by phillipthelen** [comment]
I don’t think a raspberry pi would be the right choice here. Even if possible you’d be running full blown Linux just to control a few pins. You can however add a microcontroller like the raspberry pi pico as an additional mcu and use its pins to achieve what you mentioned.

The printer already contains an arm system similar to the raspberry pi, so not sure why you would move computations from the printer to the pi. There printer already has plenty processing power.

**Reply by linhungsam** [reply]
Hi Phillip, 

You are right about the processing power of the stock motherboard. However, I dont know much about this motherboard, and there aren't much documentation/guide about it, therefore tinkering with it may have so many risks. On the other hand, there are many resources for tinkering RPi and I have experience with its GPIO input/output, working on it is so much easier for me.

Do you know any online resources about this Nep4 motherboard? something like how to add a new MCU or extension components etc...

Thanks,

Sam
